Tarea Clase 5

felipebertocchi opened the pull request:
**ignoren el nombre de la branch**

Este pull request agrega las tareas de la clase 5 nada más, las de la clase 6 las agregaré más tarde.
Modifiqué algunas cosas de las tareas, por ejemplo en la de las listas de números, dejo que el usuario genere su propia lista que se guarda dentro de un elemento "ol".

felipebertocchi committed:
Update index.html

fabertocchi committed:
Clase 6

fabertocchi committed:
Update Tarea Clase 5

fabertocchi committed:
Update index.html

fabertocchi committed:
Tarea tiempos de video

fabertocchi committed:
Update rarg-tiempo-videos.html

fabertocchi committed:
Update rarg-tiempo-videos.html

fabertocchi committed:
Modifica los nombres de la tarea

Edita los nombres la tarea de clase 5 sobre calcular el tiempo de los videos

fabertocchi committed:
Agrega la última tarea de la clase 5

fabertocchi committed:
modifica los nombres de las tareas de clase 5

fabertocchi committed:
Update tarea-tiempo-videos.html

Cuando se ingresan datos con el botón, los valores en las casillas se borran para así poder ingresar nuevos datos sin tener que borrarlos o sobreescribir.

fsodano commented on tareas/clase-5/tarea-clase-5.js:
```diff
@@ -18,10 +18,30 @@ $botonCalcular.onclick = function()
+
+$calcularSalarioMensual.onclick = function(){
+  console.log("Hiciste click!")
+  let SalarioAnual = document.querySelector("#salario-anual").value;
```
las variables van con camelCase

fsodano commented on tareas/clase-5/tarea-clase-5.js:
```diff
@@ -18,10 +18,30 @@ $botonCalcular.onclick = function()
+  console.log("Hiciste click!")
+  let SalarioAnual = document.querySelector("#salario-anual").value;
+  let SalarioMensual = calcularSalarioMensual(SalarioAnual);
+  console.log(SalarioMensual);
```
el producto final no debería tener los `console.log`, sólo para que lo tengas en cuenta a futuro cuando lo hagas profesionalmente.

fsodano commented on tareas/clase-5/tarea-listas.html:
```diff
@@ -0,0 +1,39 @@
+    <script type="text/javascript" src="tarea-listas.js"></script>
+</body>
+
+</html>
```
Siempre dejá una nueva línea al final del archivo, por un tema de compatibilidad con softwares más viejos. GitHub te advierte poniéndote ese simbolito que ves arriba.

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+const nodoLista = document.querySelector('ol');
```
seguí el mismo patrón, si a los elementos les estás poniendo `$`, por qué a este no?

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+let lista;
+
+$generarLista.onclick = function () {
+    n = Number(document.querySelector("#cantidad-numeros").value)
```
esta `n` no está siendo definida explícitamente, agregale un `const` adelante.

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+$generarLista.onclick = function () {
+    n = Number(document.querySelector("#cantidad-numeros").value)
+    for (let i = 0; i < n; i++) {
+        let numeroAleatorio = Math.round(Math.random() * 100)
```
👍

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+    for (let i = 0; i < lista.length; i++) {
+        suma += Number(lista[i].textContent)
+    }
+    const promedio = suma / lista.length;
+    return promedio;
```
Esto está bien  pero también podés hacer directamente el `return suma / lista.length;`

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+}
+
+function calcularMasPequenio(lista) {
+    let masPequenio = Number(lista[0].textContent);
```
Ojo al piojo con los casos donde no haya elementos en la  lista!

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+
+function calcularMasFrecuente(lista) {
+    let masFrecuente = Number(lista[0].textContent);
+    let mf = 1;
```
qué es `mf`? Por como lo inicializas imagino que significa algo así como `vecesRepetido`

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+function calcularMasFrecuente(lista) {
+    let masFrecuente = Number(lista[0].textContent);
+    let mf = 1;
+    var m = 0;
```
y la `m`?

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+        }
+        m = 0;
+    }
+        return (`${masFrecuente} (se repite ${mf} veces)`);
```
ojo acá con la identación, debería ser uno menos, a la misma altura que el `for`

fsodano commented on tareas/clase-5/tarea-listas.js:
```diff
@@ -0,0 +1,111 @@
+
+$calcularPromedio.onclick = function () {
+    let promedio = calcularPromedio(lista);
+    let nuevaDivision = document.createElement('div');
```
estás siempre creando un `div` nuevo, no? Es decir que vas a ver los cálculos anteriores

felipebertocchi replied:
Si, estaba viendo como separar o poner cada cosa en otras líneas... también me quedaban muy pegados, pero ya aprendí a usar los tag de br y hr
También quería ver de hacer un boton de reset que borre el texto o los div que voy haciendo, si eso es posible

fsodano commented on tareas/clase-5/tarea-nombre.html:
```diff
@@ -0,0 +1,24 @@
+</head>
+
+<body>
+    <h1 style="font-size: 50px; font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif; ">Bienvenido!</h1>
```
un gran `no-no` es poner estilos directamente en los elementos, ya tenés un `main.css` así que agregá los estilos ahí!

felipebertocchi replied:
no le metí tanta garra al CSS, voy a ver de darle una vuelta al video y ponerme a full!

fsodano commented on tareas/clase-5/tarea-nombre.js:
```diff
@@ -0,0 +1,10 @@
+const $ingresar = document.querySelector("#ingresar");
+
+$ingresar.onclick = function () {
+    const $primerNombreUsuario = document.querySelector("#primer-nombre-usuario").value;
```
Ojo que estos son values, entonces si yo leo `$primerNombreUsuario`, imagino que estás haciendo referencia a un elemento, no a un valor. En este caso sería `primerNombreUsuario`.

fsodano commented on tareas/clase-5/tarea-nombre.js:
```diff
@@ -0,0 +1,10 @@
+    const $primerNombreUsuario = document.querySelector("#primer-nombre-usuario").value;
+    const $segundoNombreUsuario = document.querySelector("#segundo-nombre-usuario").value;
+    const $apellidoUsuario = document.querySelector("#apellido-usuario").value;
+    const $nombreUsuario = $primerNombreUsuario + " " + $segundoNombreUsuario + " " + $apellidoUsuario;
```
Este es un gran momento para usar template literals!

felipebertocchi replied:
Vos sabes que lo venía usando y en este lo re pasé de largo? ahora veo de reformatear!

fsodano commented on tareas/clase-5/tarea-tiempo-videos.js:
```diff
@@ -0,0 +1,46 @@
+    nuevoParrafo.appendChild(textoParrafo);
+    nodoPagina.appendChild(nuevoParrafo);
+
+    return horas, minutos, segundos;
```
esto te devuelve sólo `segundos`, si querés devolver todo podés usar un array: `return [horas, minutos, segundos]`

felipebertocchi replied:
Si! me di cuenta tarde en otro ejercicio que solo devuelve un valor, o array las function

felipebertocchi commented:
Bueno, por como lo estoy viendo voy a tener que hacer varios cambios jaja
Me comí un monton de cosas, y otras las hice así nomás probando y me olvidé de darle un mejor formato, más presentable.
Gracias por los comentarios

felipebertocchi committed:
Update tarea-clase-5.js & index.html

Modifique según comentarios en pull request #2

felipebertocchi committed:
Crea listas.css

Archivo de CSS para darle un mejor estilo al html de la tarea de listas.

felipebertocchi committed:
Update tarea-listas

Mejora y correción de errores según comentarios en pull request #2

felipebertocchi commented:
Tarea de listas corregida:
- Agregué formato con css para poder separar la pantalla en 2, para tener un mejor paronama de los resultados.
- Agregué un "$" a los valores que toman un elemento o lo crean en html
- Corregí la función `masFrecuente`, dandole una mejor sintaxis
- Agregué varios `alert` que avisan al usuario si no ingresó correctamente la cantidad de números para generar la lista, o si quiso calcular alguna función sin haber definido la lista.

felipebertocchi commented:
Consulta sobre GitHub: 
¿Está bien la manera que voy corrigiendo los errores? Me refiero a si debería citar algún commit o tus comentarios, aunque ví que cada vez que hago un push aparece en la timeline los archivos que voy editando, pero para saber nomás.
Vi en los comentarios que pusiste que esta la opción **"Resolve conversation"**, que supongo es para cerrarlos una vez que se corrigieron los errores. Correspondería que vos los cierres, no?

felipebertocchi committed:
Update tarea-listas.js

Modifique el valor que devulve la función `calcularMasFrecuente` ya que al mostrar cuantas veces se repite algún valor daba lugar a confusiones. (i.e. un valor se repite X veces, pero aparece X+1 veces en la lista) Ahora muestra cuantas veces aparece un valor dentro de la lista.

felipebertocchi committed:
Update tarea-nombre

Modificaciones según comentarios de pull request #2

felipebertocchi committed:
Update tarea-tiempo-videos.js
